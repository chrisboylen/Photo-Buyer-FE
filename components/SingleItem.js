import React, { Component } from "react";
import gql from "graphql-tag";
import { Query } from "react-apollo";
import Error from "./ErrorMessage";
import styled from "styled-components";
import Head from "next/head";
import formatMoney from "../lib/formatMoney";

const SingleItemStyles = styled.div`
  border-radius: 5px;
  max-width: 1200px;
  margin: 1rem auto;
  box-shadow: ${props => props.theme.bs};
  display: grid;
  grid-auto-columns: 1fr;
  grid-auto-flow: column;
  min-height: 800px;
  h2 {
    background: ${props => props.theme.blue};
    border-radius: 5px 5px 0 0;
    color: ${props => props.theme.white};
    text-align: center;
    margin: 0;
    span {
      color: #c7e4ff;
    }
  }
  img {
    width: 100%;
    height: 100%;
    object-fit: contain;
    border-radius: 0 0 5px 5px;
  }
  .details {
    margin: 3rem;
    font-size: 2rem;
  }
`;

const SINGLE_ITEM_QUERY = gql`
  query SINGLE_ITEM_QUERY($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      largeImage
    }
  }
`;
class SingleItem extends Component {
  render() {
    return (
      <Query
        query={SINGLE_ITEM_QUERY}
        variables={{
          id: this.props.id
        }}
      >
        {({ error, loading, data }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          if (!data.item) return <p>No Item Found for {this.props.id}</p>;
          const item = data.item;
          return (
            <SingleItemStyles>
              <Head>
                <title>Photo Buyer | {item.title}</title>
              </Head>
              <div className="details">
                <h2>
                  <span>Viewing: </span> {item.title}
                </h2>
                <img src={item.largeImage} alt={item.title} />
                <p>{item.description}</p>
              </div>
            </SingleItemStyles>
          );
        }}
      </Query>
    );
  }
}

export default SingleItem;
export { SINGLE_ITEM_QUERY };
