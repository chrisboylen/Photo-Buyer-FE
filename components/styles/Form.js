import styled, { keyframes } from "styled-components";

const loading = keyframes`
  from {
    background-position: 0 0;
  }

  to {
    background-position: 100% 100%;
  }
`;

const Form = styled.form`
  box-shadow: 0 0 15px 6px rgba(25, 62, 93, 0.37);
  background: #9ed2ff4d;
  border-radius: 5px;
  padding: 20px;
  font-size: 1.5rem;
  line-height: 1.5;
  font-weight: 600;
  label {
    display: block;
    margin-bottom: 1rem;
  }
  input,
  textarea,
  select {
    width: 100%;
    padding: 0.5rem;
    font-size: 1rem;
    border: 1px solid black;
    border-radius: 3px;
    &:focus {
      outline: 0;
      border-color: ${props => props.theme.blue};
    }
  }
  button,
  input[type="submit"] {
    width: auto;
    background: #4dadfe;
    color: white;
    border: 0;
    border-radius: 5px;
    font-size: 2rem;
    font-weight: 600;
    padding: 0.5rem 1.2rem;
  }
  button {
    background: #ffe504;
    color: #193e5d;
  }
  fieldset {
    border: 0;
    padding: 0;

    &[disabled] {
      opacity: 0.5;
    }
    &::before {
      height: 10px;
      content: "";
      display: block;
      background-image: linear-gradient(to right, #4dadfe 0%, #193e5d 100%);
      border-radius: 100px;
    }
    &[aria-busy="true"]::before {
      background-size: 50% auto;
      animation: ${loading} 0.5s linear infinite;
    }
  }
`;

export default Form;
