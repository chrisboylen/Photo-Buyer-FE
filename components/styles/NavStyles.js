import styled from "styled-components";

const NavStyles = styled.ul`
  background: ${props => props.theme.lightblue};
  margin: 0;
  padding: 0 0 0 7rem;
  font-size: 2rem;
  .web-nav {
    display: flex;
    @media (max-width: 760px) {
      align-items: center;
      flex-direction: column;
      font-size: 3rem;
      opacity: 0;
      position: absolute;
      top: 200%;
      left: 50%;
      transform: translate(-50%, -50%);
      transition: all 0.8s ease-in-out;
      list-style: none;
      width: 100%;
      z-index: 1500;
    }
  }
  .mobile-checkbox {
    display: none;
    &:checked ~ .mobile-bkg {
      transform: scale(100);
    }
    &:checked ~ .web-nav {
      opacity: 1;
      width: 100%;
    }
    &:checked + .mobile-btn .mobile-icon {
      background: transparent;
    }
    &:checked + .mobile-btn .mobile-icon::before {
      top: 0;
      transform: rotate(135deg);
    }
    &:checked + .mobile-btn .mobile-icon::after {
      top: 0;
      transform: rotate(-135deg);
    }
  }
  .mobile-icon {
    position: relative;
    margin-top: 3.4rem;
    &,
    &::before,
    &::after {
      width: 3.5rem;
      height: 3px;
      background: ${props => props.theme.blue};
      display: inline-block;
    }
    /* &:hover,
    &:hover::before,
    &:hover::after {
      background: yellow;
    } */
    &::before,
    &::after {
      content: "";
      position: absolute;
      left: 0;
      transition: all 0.2s;
    }
    &::before {
      top: -1rem;
    }
    &::after {
      top: 1rem;
    }
  }
  .mobile-btn {
    display: none;
    background-color: #e8f4ff;
    box-shadow: 0 1rem 3rem #193e5d26;
    height: 7rem;
    width: 7rem;
    position: fixed;
    top: 3rem;
    right: 3rem;
    border-radius: 50%;
    z-index: 2000;
    text-align: center;
    cursor: pointer;
    @media (max-width: 760px) {
      display: block;
    }
    &:hover {
      background: yellow;
    }
  }
  .mobile-bkg {
    display: none;
    height: 6rem;
    width: 6rem;
    border-radius: 50%;
    position: fixed;
    top: 3.5rem;
    right: 3.5rem;
    background-image: radial-gradient(
      ${props => props.theme.lightblue},
      ${props => props.theme.blue}
    );
    z-index: 1000;
    transition: transform 0.8s cubic-bezier(0.86, 0, 0.07, 1);
    @media (max-width: 760px) {
      display: block;
    }
  }
  a,
  button {
    padding: 1rem 3rem;
    display: flex;
    align-items: center;
    position: relative;
    text-transform: uppercase;
    font-weight: 900;
    font-size: 1em;
    background: none;
    border: 0;
    cursor: pointer;
    color: ${props => props.theme.black};
    font-weight: 800;
    &:after {
      height: 2px;
      background: yellow;
      content: "";
      width: 0;
      position: absolute;
      transform: translateX(-50%);
      transition: width 0.4s;
      transition-timing-function: cubic-bezier(1, -0.65, 0, 2.31);
      left: 50%;
      margin-top: 2rem;
    }
    &:hover,
    &:focus {
      outline: none;
      color: yellow;
      &:after {
        width: calc(100% - 60px);
      }
    }
  }
  @media (max-width: 1300px) {
    width: 100%;
    font-size: 1.5rem;
  }
`;

export default NavStyles;
