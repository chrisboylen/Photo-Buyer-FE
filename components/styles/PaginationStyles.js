import styled from "styled-components";

const PaginationStyles = styled.div`
  background: #9ed2ff4d;
  text-align: center;
  display: inline-grid;
  grid-template-columns: repeat(4, auto);
  align-items: stretch;
  justify-content: center;
  align-content: center;
  margin: 2rem 0;
  border: 2px solid ${props => props.theme.blue};
  border-radius: 10px;
  min-width: 40rem;
  @media (max-width: 760px) {
    margin: 0;
  }
  & > * {
    margin: 0;
    padding: 15px 30px;
    border-right: 1px solid ${props => props.theme.lightgrey};
    &:last-child {
      border-right: 0;
    }
    @media (max-width: 760px) {
      padding: 5px 15px;
    }
  }
  a[aria-disabled="true"] {
    color: grey;
    pointer-events: none;
  }
`;

export default PaginationStyles;
