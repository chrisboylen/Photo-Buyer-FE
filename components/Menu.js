import React, { Component } from "react";
import Link from "next/link";
import { MobileStyles } from "./styles/NavStyles";
import User from "../components/User";
import Signout from "./Signout";
import { Mutation } from "react-apollo";
import { TOGGLE_CART_MUTATION } from "./Cart";
import CartCount from "./CartCount";
import PropTypes from "prop-types";

class Menu extends Component {
  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <MobileStyles data-test="mobile">
            <Link href="/items">
              <a data-test="items" onClick={this.props.toggleMenu}>
                Shop
              </a>
            </Link>
            {me && (
              <>
                <Link href="/sell">
                  <a onClick={this.props.toggleMenu}>Sell</a>
                </Link>
                <Link href="/orders">
                  <a onClick={this.props.toggleMenu}>Orders</a>
                </Link>
                <Signout />
                <Mutation mutation={TOGGLE_CART_MUTATION}>
                  {toggleCart => (
                    <button onClick={toggleCart} data-test="signout">
                      My Cart
                      <CartCount
                        count={me.cart.reduce(
                          (tally, cartItem) => tally + cartItem.quantity,
                          0
                        )}
                      />
                    </button>
                  )}
                </Mutation>
              </>
            )}
            {!me && (
              <>
                <Link href="/signup">
                  <a onClick={this.props.toggleMenu}>Signup</a>
                </Link>
                <Link href="/signin">
                  <a onClick={this.props.toggleMenu}>Signin</a>
                </Link>
              </>
            )}
          </MobileStyles>
        )}
      </User>
    );
  }
}

Menu.propTypes = {};

export default Menu;
