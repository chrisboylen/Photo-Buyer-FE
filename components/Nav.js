import React, { Component } from "react";
import Link from "next/link";
import { NavStyles, MobileStyles } from "./styles/NavStyles";
import User from "./User";
import Signout from "./Signout";
import { Mutation } from "react-apollo";
import { TOGGLE_CART_MUTATION } from "./Cart";
import CartCount from "./CartCount";
import Menu from "./Menu";

class Nav extends Component {
  state = {
    isOpen: false
  };

  toggleMenu = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <NavStyles data-test="nav">
            {this.state.isOpen && (
              <Menu isOpen={this.state.isOpen} toggleMenu={this.toggleMenu} />
            )}
            <button
              onClick={this.toggleMenu}
              className="mobile-checkbox"
              id="mobile-toggle"
            />
            <label htmlFor="mobile-toggle" className="mobile-btn">
              <span
                className={
                  this.state.isOpen
                    ? "mobile-icon mobile-icon-open"
                    : "mobile-icon"
                }
              >
                &nbsp;
              </span>
            </label>
            <div className="mobile-bkg">&nbsp;</div>
            <div className="web-nav">
              <Link href="/items">
                <a>Shop</a>
              </Link>
              {me && (
                <>
                  <Link href="/sell">
                    <a>Sell</a>
                  </Link>
                  <Link href="/orders">
                    <a>Orders</a>
                  </Link>
                  <Signout />
                  <Mutation mutation={TOGGLE_CART_MUTATION}>
                    {toggleCart => (
                      <button onClick={toggleCart}>
                        My Cart
                        <CartCount
                          count={me.cart.reduce(
                            (tally, cartItem) => tally + cartItem.quantity,
                            0
                          )}
                        />
                      </button>
                    )}
                  </Mutation>
                </>
              )}
              {!me && (
                <>
                  <Link href="/signup">
                    <a>Signup</a>
                  </Link>
                  <Link href="/signin">
                    <a>Signin</a>
                  </Link>
                </>
              )}
            </div>
          </NavStyles>
        )}
      </User>
    );
  }
}

export default Nav;
