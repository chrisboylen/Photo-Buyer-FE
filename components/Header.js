import Link from "next/link";
import styled from "styled-components";
import Nav from "./Nav";
import Router from "next/router";
import NProgress from "nprogress";
import Cart from "./Cart";
import Search from "./Search";

Router.onRouteChangeStart = () => {
  NProgress.start();
};
Router.onRouteChangeComplete = () => {
  NProgress.done();
};
Router.onRouteChangeError = () => {
  NProgress.done();
};

const Logo = styled.h1`
  background: ${props => props.theme.lightblue};
  font-size: 8rem;
  margin: 0;
  min-height: 12rem;
  position: relative;
  z-index: 1500;
  a {
    margin: 0 0 0 10rem;
    padding: 1rem 2rem 0.6rem 2rem;
    border-radius: 8px;
    background: #e8f4ff;
    color: #4dadfe;
    text-transform: uppercase;
    text-decoration: none;
  }
  span {
    background: linear-gradient(to right, #00f1fe 0%, #4dadfe 100%);
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
  img {
    width: 6rem;
    margin: 0 2rem 0 0;
  }
  @media (max-width: 1300px) {
    font-size: 6rem;
    img {
      width: 4.5rem;
    }
    @media (max-width: 760px) {
      line-height: 1.5;
      a {
        top: 2rem;
        left: 3rem;
        margin: 0;
        padding: 0.3rem 1.1rem 0 1.1rem;
        position: relative;
      }
      img {
        margin: 0;
        width: 4rem;
      }
      span {
        display: none;
      }
    }
  }
`;

const StyledHeader = styled.header`
  left: 0;
  top: 0;
  position: sticky;
  z-index: 500;
  .bar {
    grid-template-columns: auto 1fr;
    justify-content: space-between;
    align-items: stretch;
    @media (max-width: 1300px) {
      grid-template-columns: 1fr;
      justify-content: center;
    }
  }
  .sub-bar {
    background: linear-gradient(
      to bottom,
      #8cbce2 0%,
      #8cbce2 70%,
      #193e5d 100%
    );
    display: grid;
    grid-template-columns: 1fr;
    border-bottom: 1px solid ${props => props.theme.lightgrey};
  }
`;

const Header = () => (
  <StyledHeader>
    <div className="bar">
      <Logo>
        <Link href="/">
          <a>
            <img src="../static/camera.png" alt="Camera Icon" />
            <span className="text">Photo Buyer</span>
          </a>
        </Link>
      </Logo>
      <Nav />
    </div>
    <div className="sub-bar">
      <Search />
    </div>
    <Cart />
  </StyledHeader>
);

export default Header;
