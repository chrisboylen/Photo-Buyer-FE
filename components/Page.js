import React, { Component } from "react";
import styled, { ThemeProvider, injectGlobal } from "styled-components";
import Header from "./Header";
import Meta from "./Meta";

const theme = {
  lightblue: "#9ed2ff",
  blue: "#4dadfe",
  black: "#193e5d",
  grey: "#3a3a3a",
  lightgrey: "#b1c0e0",
  white: "#e8f4ff",
  offWhite: "#ededed",
  maxWidth: "1000px",
  bs: "0 12px 24px 0 rgba(0, 0, 0, 0.21)"
};

const StyledPage = styled.div`
  color: ${props => props.theme.black};
`;

const Inner = styled.div`
  max-width: ${props => props.theme.maxWidth};
  margin: 0 auto;
  padding: 2rem;
  @media (max-width: 760px) {
    padding: 0;
  }
`;

injectGlobal`
  @font-face {
    font-family: 'DoHyeon-Regular';
    src: url('/static/DoHyeon-Regular.ttf')
    format('woff2');
    font-weight: normal;
    font-style: normal;
  }
  html {
    box-sizing: border-box;
    font-size: 10px;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    background: #e8f4ff52;
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: 'DoHyeon-Regular';
  }
  a {
    text-decoration: none;
    color: ${theme.black}
  }
`;

class Page extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <StyledPage>
          <Meta />
          <Header />
          <Inner>{this.props.children}</Inner>
        </StyledPage>
      </ThemeProvider>
    );
  }
}

export default Page;
