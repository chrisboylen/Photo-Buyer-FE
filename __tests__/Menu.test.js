import { mount } from "enzyme";
import wait from "waait";
import toJSON from "enzyme-to-json";
import { MockedProvider } from "react-apollo/test-utils";
import { ApolloConsumer } from "react-apollo";
import { CURRENT_USER_QUERY } from "../components/User";
import { fakeUser } from "../lib/testUtils";
import Menu from "../components/Menu";
import { MobileStyles } from "../components/styles/NavStyles";

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: null } }
  }
];

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: fakeUser() } }
  }
];

describe("Menu", () => {
  let wrapper;
  let toggleMenuMock;

  beforeEach(() => {
    wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <Menu toggleMenu={toggleMenuMock} />
      </MockedProvider>
    );
    toggleMenuMock = jest.fn();
  });

  it("renders and matches snapshot when signed in", async () => {
    await wait();
    wrapper.update();

    expect(toJSON(wrapper.find('ul[data-test="mobile"]'))).toMatchSnapshot();
    expect(wrapper.find('ul[data-test="mobile"]').children().length).toBe(5);
    expect(wrapper.find('ul[data-test="mobile"]').text()).toContain("Signout");
  });

  it("renders and matches snapshot when not signed in", async () => {
    wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <Menu toggleMenu={toggleMenuMock} />
      </MockedProvider>
    );
    await wait();
    wrapper.update();

    expect(toJSON(wrapper.find('ul[data-test="mobile"]'))).toMatchSnapshot();
  });
});
