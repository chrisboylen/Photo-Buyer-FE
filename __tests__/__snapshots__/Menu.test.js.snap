// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders and matches snapshot when not signed in 1`] = `
<ul
  className="NavStyles__MobileStyles-sc-10phz6m-1 hFHAXc"
  data-test="mobile"
>
  <Link
    href="/items"
  >
    <a
      data-test="items"
      href="/items"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
  <Link
    href="/signup"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
  </Link>
  <Link
    href="/signin"
  >
    <a
      href="/signin"
      onClick={[Function]}
    >
      Signin
    </a>
  </Link>
</ul>
`;

exports[`Menu renders and matches snapshot when signed in 1`] = `
<ul
  className="NavStyles__MobileStyles-sc-10phz6m-1 hFHAXc"
  data-test="mobile"
>
  <Link
    href="/items"
  >
    <a
      data-test="items"
      href="/items"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
  <Link
    href="/sell"
  >
    <a
      href="/sell"
      onClick={[Function]}
    >
      Sell
    </a>
  </Link>
  <Link
    href="/orders"
  >
    <a
      href="/orders"
      onClick={[Function]}
    >
      Orders
    </a>
  </Link>
  <Signout>
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "SIGN_OUT_MUTATION",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "signout",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "message",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 70,
            "start": 0,
          },
        }
      }
      refetchQueries={
        Array [
          Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "me",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "permissions",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "orders",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cart",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "quantity",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "item",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "price",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "image",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "title",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "description",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 271,
                "start": 0,
              },
            },
          },
        ]
      }
    >
      <button
        onClick={[Function]}
      >
        Signout
      </button>
    </Mutation>
  </Signout>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [
                    Object {
                      "arguments": Array [],
                      "kind": "Directive",
                      "name": Object {
                        "kind": "Name",
                        "value": "client",
                      },
                    },
                  ],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "toggleCart",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 41,
          "start": 0,
        },
      }
    }
  >
    <button
      data-test="signout"
      onClick={[Function]}
    >
      My Cart
      <CartCount
        count={0}
      >
        <CartCount__AnimationStyles>
          <span
            className="CartCount__AnimationStyles-zy5sf2-0 goAznk"
          >
            <TransitionGroup
              childFactory={[Function]}
              component="div"
            >
              <div>
                <CSSTransition
                  className="count"
                  classNames="count"
                  in={true}
                  key=".$0"
                  onExited={[Function]}
                  timeout={
                    Object {
                      "enter": 400,
                      "exit": 400,
                    }
                  }
                  unmountOnExit={true}
                >
                  <Transition
                    appear={false}
                    className="count"
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={
                      Object {
                        "enter": 400,
                        "exit": 400,
                      }
                    }
                    unmountOnExit={true}
                  >
                    <CartCount__Dot
                      className="count"
                    >
                      <div
                        className="count CartCount__Dot-zy5sf2-1 krogjt"
                      >
                        0
                      </div>
                    </CartCount__Dot>
                  </Transition>
                </CSSTransition>
              </div>
            </TransitionGroup>
          </span>
        </CartCount__AnimationStyles>
      </CartCount>
    </button>
  </Mutation>
</ul>
`;
